papaparse.js

```diff
@@ -1040,10 +1040,14 @@
 			return _results;
 		}
 
-		function guessDelimiter(input, newline)
+		function guessDelimiter(input, newline, customDelimChoices)
 		{
-			var delimChoices = [',', '\t', '|', ';', Papa.RECORD_SEP, Papa.UNIT_SEP];
-			var bestDelim, bestDelta, fieldCountPrevRow;
+			var bestDelim, bestDelta, delimChoices, fieldCountPrevRow;
+
+			if (!customDelimChoices)
+				delimChoices = [',', '\t', '|', ';', Papa.RECORD_SEP, Papa.UNIT_SEP];
+			else
+				delimChoices = customDelimChoices;
 
```

Comment:
About performance hit, looking at this line and above, it's safe to say that won't have any!
Unless the custom array would be a very large array and the file delimiter is one of the last from this array.